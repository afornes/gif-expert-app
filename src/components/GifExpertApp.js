import { useState } from "react";
import AddCategory from "./AddCategory";
import GifGrid from "./GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {

    //const [categories, setCategories] = useState(['Re: Zero']);
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr />

            <ol>
                {
                    categories.map( item => ( 
                        <GifGrid 
                            key={item}
                            category={item}
                        />
                    ))
                }
            </ol>
        </>
    );
}

export default GifExpertApp;