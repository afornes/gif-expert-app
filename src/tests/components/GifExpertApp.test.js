import { shallow } from 'enzyme';
import GifExpertApp from '../../components/GifExpertApp';

describe('Tests for Gif Expert App component', () => { 
    
    test('should show app correctly', () => { 

        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
     });

     test('should show a list of categories', () => { 
         
        const categories = ['Dragon Ball', 'Kimetsu no yaiba'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
      });
 });