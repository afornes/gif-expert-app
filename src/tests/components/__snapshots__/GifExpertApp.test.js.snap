// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Gif Expert App component should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Kimetsu no yaiba"
      key="Kimetsu no yaiba"
    />
  </ol>
</Fragment>
`;

exports[`Tests for Gif Expert App component should show app correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
